python/lsst/drp/tasks/single_frame_detect_and_measure.py

```diff
@@ -0,0 +1,298 @@
+        name="single_visit_star_reprocessed_footprints",
+        storageClass="SourceCatalog",
+        dimensions=["instrument", "visit", "detector"],
+    )
+    background = connectionTypes.Output(
+        doc=(
+            "Total background model including new detections in this task. "
+            "Note that the background model has units of ADU, while the corresponding "
+            "image has units of nJy - the image must be 'uncalibrated' before the background "
+            "can be restored."
+        ),
+        name="preliminary_visit_image_reprocessed_background",
+        dimensions=("instrument", "visit", "detector"),
+        storageClass="Background",
+    )
+
+
+class SingleFrameDetectAndMeasureConfig(
+    pipeBase.PipelineTaskConfig, pipelineConnections=SingleFrameDetectAndMeasureConnections
+):
+    # To generate catalog ids consistently across subtasks.
+    id_generator = lsst.meas.base.DetectorVisitIdGeneratorConfig.make_field()
+
+    detection = pexConfig.ConfigurableField(
+        target=lsst.meas.algorithms.SourceDetectionTask,
+        doc="Task to detect sources to return in the output catalog.",
+    )
+    sky_sources = pexConfig.ConfigurableField(
+        target=lsst.meas.algorithms.SkyObjectsTask,
+        doc="Task to generate sky sources ('empty' regions where there are no detections).",
+    )
+    deblend = pexConfig.ConfigurableField(
+        target=lsst.meas.deblender.SourceDeblendTask,
+        doc="Task to split blended sources into their components.",
+    )
+    measurement = pexConfig.ConfigurableField(
+        target=lsst.meas.base.SingleFrameMeasurementTask,
+        doc="Task to measure sources to return in the output catalog.",
+    )
+    normalized_calibration_flux = pexConfig.ConfigurableField(
+        target=lsst.meas.algorithms.NormalizedCalibrationFluxTask,
+        doc="Task to normalize the calibration flux (e.g. compensated tophats).",
+    )
+    apply_aperture_correction = pexConfig.ConfigurableField(
+        target=lsst.meas.base.ApplyApCorrTask,
+        doc="Task to apply aperture corrections to the measured sources.",
+    )
+    set_primary_flags = pexConfig.ConfigurableField(
+        target=lsst.meas.algorithms.setPrimaryFlags.SetPrimaryFlagsTask,
+        doc="Task to add isPrimary to the catalog.",
+    )
+    catalog_calculation = pexConfig.ConfigurableField(
+        target=lsst.meas.base.CatalogCalculationTask,
+        doc="Task to compute catalog values using only the catalog entries.",
+    )
+    do_add_sky_sources = pexConfig.Field(
+        dtype=bool,
+        default=True,
+        doc="Generate sky sources?",
+    )
+
+    def setDefaults(self):
+        super().setDefaults()
+
+        # Re-estimate the background
+        self.detection.reEstimateBackground = True
+        self.detection.doTempLocalBackground = False
+
+        # NOTE: these apertures were selected for HSC, and may not be
```

Comment:
I know this is also a comment on `ReprocessVisitImageTask`, but would it be worth making this a todo with an investigation ticket?